File: android-app/transfer-api/src/main/java/com/hlrms/transferapi/dto/TransferHistoryResponse.java
package com.hlrms.transferapi.dto;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

public record TransferHistoryResponse(
    UUID id,
    UUID requestId,
    String direction,
    String counterpartName,
    String counterpartTransferId,
    BigDecimal amount,
    String currency,
    String status,
    Instant createdAt
) {
}

File: android-app/app/src/main/java/com/hlrms/mobile/data/remote/transfer/TransferHistoryResponse.kt
package com.hlrms.mobile.data.remote.transfer

data class TransferHistoryResponse(

    val id: String,

    val requestId: String,

    val direction: String,

    val counterpartName: String,

    val counterpartTransferId: String,

    val amount: String,

    val currency: String,

    val status: String,

    val createdAt: String?
)
